apps/recnet-api/src/modules/rec/rec.controller.ts

```diff
@@ -63,10 +63,16 @@ export class RecController {
     summary: "Get a single rec",
     description: "Get a single rec by id.",
   })
+  @ApiBearerAuth()
   @ApiOkResponse({ type: GetRecResponse })
   @Get("rec/:id")
-  public async getRec(@Param("id") id: string): Promise<GetRecResponse> {
-    return this.recService.getRec(id);
+  @AuthOptional()
+  public async getRec(
+    @Param("id") id: string,
+    @UserOptional() authUser: AuthOptionalUser
+  ): Promise<GetRecResponse> {
+    const authUserId = authUser ? authUser.userId : null;
```

Comment:
nit: you can use nullish coalescing here
```suggestion
    const authUserId = authUser?.userId ?? null;
```